tools/resource-gen/main.go

```diff
@@ -19,6 +19,126 @@ import (
 	_ "github.com/kumahq/kuma/api/mesh/v1alpha1"
 )
 
+// NativeApuTemplate for creating a Kuma resource.
+var NativeApiTemplate = template.Must(template.New("resource").Parse(`
+// Generated by tools/resource-gen
+// Run "make generate" to update this file.
+
+{{ $pkg := printf "%s_proto" .Package }}
+{{ $tk := "` + "`" + `" }}
```

Comment:
This is making my head hurt .. what's going on here?

Reply:
[It's impossible to escape backticks inside a string literal](https://github.com/golang/go/issues/24475) + Go relying so heavily on writing code generation via string templating :shrug: .
Seems like a good way to solve this.

Reply:
OK, I see the first "`" closes the string that started at `Parse`, then we concat a double-quotes backtick, then we concat the remainder of the parse string.

I think it would be simpler to add a `.Backtick` member or function to the template context :)

Reply:
@jpeach I think pulling it out into a part of the code where the escaping is more readable is a good idea. I'm going to copy it back into short-named var like `tk` in the template though because `{{.Backtick}}` is just a lot to put into every place where a backtick usually goes.

Reply:
Could we just work around this by having all templates in a folder as raw text and use goembed to load them?

Reply:
Going to do more with templates in a subsequent PR - autogen the rest of the helpers, add new autogen type, move templates into their own files so that backticks don't need special handling. Will open issues if I don't get to it today.